scripts/copyFilesUtils.mjs

```diff
@@ -156,13 +156,18 @@ export async function createPackageFile(useEsmExports = false) {
     ...createExportFor('./*', { [srcCondition]: './src/*/index.ts' }),
   };
 
+  if (!packageDataOther.exports?.['./*']) {
+    // From the default wildcard we should exclude the esm and modern builds.
+    // If you override the wildcard, you're on your own
+    Object.assign(packageExports, {
+      ...createExportFor('./esm', null),
+      ...createExportFor('./modern', null),
+    });
+  }
```

Comment:
I don't quite get what this means specifically.

Is it `IF './*' NOT defined DISALLOW './esm' imports`? Shouldn't this be disallowed for all configs regardless of wildcard? 🤔

Reply:
I just wanted to leave the developer in control. If they override the exports, we assume they know what they're doing. e.g. icons doesn't have modern exports, we shouldn't just add a `"./modern": null` therefore.

Reply:
> Doesn't look like it would matter, as ./* and ./modern|esm are unrelated.

the `./modern` override is there because `./*` would otherwise match `./modern/index.js`, that's how they are related.

> I could override `./*`. But do nothing with `modern/esm`

yes, when you deviate from what this script prescribes, we give you full control. meaning: we won't override any paths that you expect to be working.

> Should we check if ./modern|esm are set and warn about it or skip this change?

If they are set, they will override the default of `null` in the subsequent lines.

Reply:
Thanks for the info. I'm not familiar with this script 🙃

Reply:
I'm going to rework it a bit to better reflect the intention